R/estimateParams.R

```diff
@@ -53,3 +53,68 @@ prepareData <- function(data) {
 
   as.data.frame(dataNew[, -which(names(dataNew) == "time")], row.names = seq_len(nrow(dataNew)))
 }
+
+# transition <- weibull_transition(h01 = 0.5, h02 = 1.3, h12 = 1.1, p01 = 1.2, p02 = 0.5, p12 = 1.4)
+# trial <- getOneClinicalTrial(nPat = c(5000),
+#                             transitionByArm = list(transition),
+#                             dropout = list(rate = 0.3, time = 1),
+#                             accrual = list(param = "intensity", value = 100))
+
+# estimateParams(trial, "Weibull")
+
+NegLogLik <- function(data, params, family = c("exponential", "Weibull")) {
+  family <- match.arg(family)
+
+  with(data, -sum(log(haz(exit, trans, params, family)^status * SurvTrans(exit, trans, params, family) / SurvTrans(entry, trans, params, family))))
+}
+
+haz <- function(t, trans, params, family = c("exponential", "Weibull")) {
```

Comment:
so one option would be to 
1) omit the `family` argument
2) make this an S3 generic function
3) let `piecewise_exponential` return with two classes: "TransitionParameters" as now, but also "ExponentialTransition"
4) define methods for "ExponentialTransition" and "WeibullTransition"

Reply:
I don't quite understand what you mean with 3. and 4. If I define separate methods for haz(), what object do I use to identify the correct class? Currently, we have a class "TransitionParameters" that contains the distribution name in $family.

Reply:
you could have a vector of 2 class names as the `class` attribute (we are using S3 here) - one would be more general (TransitionParameters) and one would be more specific (Exponential vs. Weibull Transition). 
yeah there would be some duplication of information with the `family` element... so it might need a larger change in the package

Reply:
Ok, I see what you mean. I guess I should be able to replace the `family` element with the additional class attribute, and then simply change the ifelse-statements throughout the package that use `family` with `class(x)`?

Reply:
yes I think so, and in some cases it would then be nicer to later replace the ifelse statements with S3 methods